### ISSUE TITLE
Move bundles to v0.2

### ISSUE THREAD

**Issue by loosebazooka** [sigstore/sigstore-conformance#105]
Just thinking about this problem since protobuf-specs says about v0.1 bundles: https://github.com/sigstore/protobuf-specs/blob/main/protos/sigstore_bundle.proto#L62

There aren't really too many compatibility issues but maybe v0.1 support should be optional for clients in conformance tests?

**Comment by haydentherapper** [comment]
Agreed. We should have some tests for v1 for clients that care, but I don’t think we should mandate v1 support.

**Comment by woodruffw** [comment]
Fully agreed as well!

**Comment by woodruffw** [comment]
#109 starts on this as a bugfix, but others need to be moved as well.

**Comment by segiddins** [comment]
I think we also need a v0.3 test as well

**Comment by loosebazooka** [comment]
@loosebazooka see if we have all bundles types, perhaps add explicit checks for all versions.

**Comment by loosebazooka** [comment]
It does look like we need a happy path for each bundle version.
`happy-path-v0.1` ...

we do have all bundle types in here, but they're spread out randomly, and some not even in passing tests.

<details open>

<summary> expand/collapse </summary>

| bundle version | test name |
| -------- | -------- |
| vnd.dev.sigstore.bundle+json;version=0.1  | bundle-empty-certificate-chain_fail |
|                                           | bundle-from-wrong-instance_fail |
|                                           | bundle-invalid-base64-signature_fail |
|                                           | bundle-malformed-json_fail |
|                                           | bundle-negative-log-index_fail |
|                                           | bundle-with-root-cert_fail |
|                                           | checkpoint-bad-keyhint_fail |
|                                           | checkpoint-wrong-roothash_fail |
|                                           | happy-path |
|                                           | incorrect-public-key_fail |
|                                           | integrated-time-in-future_fail |
|                                           | invalid-checkpoint-signature_fail |
|                                           | invalid-ct-key_fail |
|                                           | signature-mismatch_fail |
|                                           | wrong-material_fail |
| vnd.dev.sigstore.bundle+json;version=0.2  | intoto-expired-certificate_fail |
|                                           | intoto-log-entry-mismatch_fail |
|                                           | intoto-missing-inclusion-proof_fail |
|                                           | intoto-set-outside-signing-cert-validity_fail |
|                                           | intoto-tsa-timestamp-outside-cert-validity_fail |
|                                           | intoto-with-custom-trust-root |
|                                           | invalid-inclusion-proof_fail |
| vnd.dev.sigstore.bundle+json;version=0.3  | happy-path-v0.3 |
|                                           | inclusion-proof-corrupted-hash_fail |
|                                           | message-digest-mismatch_fail |
| vnd.dev.sigstore.bundle+json;version=99.9 | bundle-unknown-version_fail |
| vnd.dev.sigstore.bundle.v0.3+json         | bundle-with-sct-with-extensions |
|                                           | dsse-invalid-sig_fail |
|                                           | dsse-mismatch-envelope_fail |
|                                           | dsse-mismatch-sig_fail |
|                                           | happy-path-intoto-in-dsse-v3 |
|                                           | managed-key-and-trusted-root |
|                                           | managed-key-happy-path |
|                                           | managed-key-no-key_fail |
|                                           | managed-key-wrong-key_fail |
|                                           | rekor2-checkpoint-cosigned |
|                                           | rekor2-checkpoint-missing-log-signature_fail |
|                                           | rekor2-checkpoint-missing-origin_fail |
|                                           | rekor2-checkpoint-missing-root-hash_fail |
|                                           | rekor2-checkpoint-missing-size_fail |
|                                           | rekor2-checkpoint-multiple-cosigs |
|                                           | rekor2-checkpoint-no-matching-signature_fail |
|                                           | rekor2-checkpoint-origin-not-first |
|                                           | rekor2-checkpoint-two-sigs-cosigned |
|                                           | rekor2-checkpoint-two-sigs-from-origin |
|                                           | rekor2-dsse-happy-path |
|                                           | rekor2-dsse-invalid-sig_fail |
|                                           | rekor2-dsse-mismatch-envelope_fail |
|                                           | rekor2-dsse-mismatch-sig_fail |
|                                           | rekor2-happy-path |
|                                           | rekor2-no-inclusion-proof_fail |
|                                           | rekor2-no-timestamp_fail |
|                                           | rekor2-timestamp-outside-trust-root-tsa-validity_fail |
|                                           | rekor2-timestamp-outside-tsa-cert-validity_fail |
|                                           | rekor2-timestamp-payload-mismatch_fail |
|                                           | rekor2-timestamp-untrusted-tsa-with-embedded-cert_fail |
|                                           | rekor2-timestamp-untrusted-tsa-without-embedded-cert_fail |
|                                           | rekor2-timestamp-with-embedded-cert |
|                                           | rekor2-timestamp-with-expired-cert-chain |
|                                           | rekor2-timestamp-with-incorrect-time_fail |
|                                           | rekor2-timestamp-without-embedded-cert |
|                                           | wrong-hashedrekord-artifact_fail |
|                                           | wrong-hashedrekord-cert-and-sig_fail |
|                                           | wrong-hashedrekord-entry_fail |

</details>
